File: generated-sources/scala-finch/mojang-api/src/main/scala/com/github/asyncmc/mojang/api/scala/finch/model/SecurityChallenge.scala
package com.github.asyncmc.mojang.api.scala.finch.model

import io.circe._
import io.finch.circe._
import io.circe.generic.semiauto._
import io.circe.java8.time._
import org.openapitools._
import com.github.asyncmc.mojang.api.scala.finch.model.SecurityAnswerId
import com.github.asyncmc.mojang.api.scala.finch.model.SecurityQuestion

/**
 * Binds a question to an answer id, the answer must respect the answer id.
 * @param question 
 * @param answer 
 */
case class SecurityChallenge(question: SecurityQuestion,
                answer: SecurityAnswerId
                )

object SecurityChallenge {
    /**
     * Creates the codec for converting SecurityChallenge from and to JSON.
     */
    implicit val decoder: Decoder[SecurityChallenge] = deriveDecoder
    implicit val encoder: ObjectEncoder[SecurityChallenge] = deriveEncoder
}

File: generated-sources/javascript-closure-angular/mojang-api/API/Client/SecurityChallenge.js
goog.provide('API.Client.SecurityChallenge');

/**
 * Binds a question to an answer id, the answer must respect the answer id.
 * @record
 */
API.Client.SecurityChallenge = function() {}

/**
 * @type {!API.Client.SecurityQuestion}
 * @export
 */
API.Client.SecurityChallenge.prototype.question;

/**
 * @type {!API.Client.SecurityAnswerId}
 * @export
 */
API.Client.SecurityChallenge.prototype.answer;

